x-pack/platform/test/fleet_api_integration/apis/policy_secrets.ts

```diff
@@ -128,7 +128,9 @@ export default function (providerContext: FtrProviderContext) {
       });
     };
 
-    const cleanupAgents = async () => {
+    const cleanupAgents = async (attempt: number = 0) => {
+      const maxAttempts = 3;
```

Comment:
we've used `pRetry` for similar retry operations in the past, might be worth taking a look to see if you want to use that here